pkg/power/components/source/rapl_sysfs_util.go

```diff
@@ -19,44 +19,20 @@ package source
 import (
 	"fmt"
 	"os"
+	"os/exec"
 	"strconv"
 	"strings"
-
-	"k8s.io/klog/v2"
 )
 
-func getNumCPUs() int {
-	data, err := os.ReadFile(cpuInfoPath)
-	if err != nil {
-		klog.V(2).Infoln(err)
-	}
-	return strings.Count(string(data), "processor")
-}
-
 func getNumPackage() int {
-	var numPackage int
-	pkgMap := map[int]bool{}
-	numCPUs := getNumCPUs()
-	for i := 0; i < numCPUs; i++ {
-		path := fmt.Sprintf(numPkgPathTemplate, i)
-		data, err := os.ReadFile(path)
-		if err != nil {
-			break
-		}
-
-		if id, err := strconv.Atoi(strings.TrimSpace(string(data))); err == nil {
-			if _, exist := pkgMap[id]; !exist {
-				numPackage++
-				pkgMap[id] = true
-			}
-		}
-	}
-	return numPackage
+	out, _ := exec.Command("bash", "-c", "lscpu |grep Socket").Output()
```

Comment:
We should avoid run bash command from the code since it is not a good design practice.

Reply:
Here is an alternative implementation

```golang
func getNumPackage() int {
  sockets := make(map[int]bool)
  file, err := os.Open("/proc/cpuinfo")
  if err != nil {
      fmt.Println(err)
      return 0
  }
  defer file.Close()

  scanner := bufio.NewScanner(file)
  for scanner.Scan() {
      line := scanner.Text()
      if strings.Contains(line, "physical id") {
          substrings := strings.Split(line, ":")
          socket, err := strconv.Atoi(strings.TrimSpace(substrings[1]))
          if err != nil {
              fmt.Println("Error:", err)
              return 0
          }
          sockets[socket] = true
      }
  }

  if err := scanner.Err(); err != nil {
      fmt.Println(err)
      return 0
  }
  
  return len(sockets)
}
```

Reply:
Thanks @marceloamaral  for pointing out his, yes, this alternative implementation scans single file instead of opens N topology files, the for loop times are same. If we choose this way, maybe we could use single function to both count core number and package number.

Reply:
Yes, we should count both in only one loop.